composer/callbacks/image_monitor.py

```diff
@@ -0,0 +1,194 @@
+        try:
+            self.interval = Time.from_timestring(interval)
+        except ValueError as error:
+            raise ValueError(f'Invalid time string for parameter interval') from error
+
+        # Verify that the interval has supported units.
+        if self.interval.unit not in [TimeUnit.BATCH, TimeUnit.EPOCH]:
+            raise ValueError(f'Invalid time unit for parameter interval: '
+                             f'{self.interval.unit}')
+
+    def _log_inputs(self, state: State, logger: Logger, key: str):
+        inputs = state.batch_get_item(key=self.input_key)
+        if not isinstance(inputs, torch.Tensor):
+            raise NotImplementedError('Multiple input tensors not supported yet')
+        # Verify inputs is a valid shape for conversion to an image
+        if _check_for_image_format(inputs):
+            table = _make_input_images(inputs, self.num_images)
+            logger.data_batch({key: table})
+
+    def _log_segmented_inputs(self, state: State, logger: Logger, key: str):
+        inputs = state.batch_get_item(key=self.input_key)
+        targets = state.batch_get_item(key=self.target_key)
+        outputs = state.outputs
+
+        if not isinstance(inputs, torch.Tensor):
+            raise NotImplementedError('Multiple input tensors not supported yet')
+        if not isinstance(targets, torch.Tensor):
+            raise NotImplementedError('Multiple target tensors not supported yet')
+        if not isinstance(outputs, torch.Tensor):
+            raise NotImplementedError('Multiple output tensors not supported yet')
+
+        table = _make_segmentation_images(inputs, targets, outputs, self.num_images, self.shift_targets)
+        logger.data_batch({key: table})
+
+    def before_forward(self, state: State, logger: Logger):
+        assert isinstance(self.interval, Time)
+        if self.mode.lower() == 'input' and state.timestamp.get(self.interval.unit).value % self.interval.value == 0:
+            self._log_inputs(state, logger, 'Images/Train')
+
+    def eval_before_forward(self, state: State, logger: Logger):
+        assert isinstance(self.interval, Time)
+        if self.mode.lower() == 'input' and state.eval_timestamp.get(TimeUnit.BATCH).value == 0:
+            self._log_inputs(state, logger, 'Images/Eval')
+
+    def before_loss(self, state: State, logger: Logger):
+        assert isinstance(self.interval, Time)
+        if self.mode.lower() == 'segmentation' and state.timestamp.get(
+                self.interval.unit).value % self.interval.value == 0:
+            self._log_segmented_inputs(state, logger, 'Images/Train')
+
+    def eval_after_forward(self, state: State, logger: Logger):
+        assert isinstance(self.interval, Time)
+        if self.mode.lower() == 'segmentation' and state.eval_timestamp.get(TimeUnit.BATCH).value == 0:
+            self._log_segmented_inputs(state, logger, 'Images/Eval')
+
+
+def _make_input_images(inputs: torch.Tensor, num_images: int):
```

Comment:
you could change the inputs type to `Union[torch.Tensor, np.ndarray, PIL.Image]` and do some `if isinstance` or `try... except` logic and then cast it to `torch.tensor`.

Reply:
Ok, I added a check for numpy arrays and PIL images, so at least the appropriate error should get raised. I'm going to hold off on support for these types for a future PR.